source/material_model/latent_heat.cc

```diff
@@ -267,41 +156,45 @@ namespace aspect
             double entropy_gradient_pressure = 0.0;
             double entropy_gradient_temperature = 0.0;
             const double rho = out.densities[i];
-            unsigned int number_of_phase_transitions;
-
-            // transition defined by depth
-            if (use_depth)
-              number_of_phase_transitions= transition_depths.size();
-            // transition defined by pressure
-            else
-              number_of_phase_transitions= transition_pressures.size();
 
+            // Number of phase transitions
+            const unsigned int number_of_phase_transitions = phase_function.n_phase_transitions();
```

Comment:
Is there a reason to declare this twice?
I think it would be better have this variabe once and then reuse it. Maybe even remove the {} that are not needed (like in lines 73 and 152; and 155 and 200) altogether, that would fix the problem.